File: TimeValidator/TimeValidator.kt
fun isValidTime(time: String): Boolean {
    // Проверяем, что длина строки равна 4
    if (time.length != 4) {
        return false // Должно быть ровно 4 цифры
    }

    // Первые 2 символа - часы, последние 2 символа - минуты
    val hours = time.substring(0, 2)
    val minutes = time.substring(2, 4)

    // Преобразуем строки в числа
    val h = hours.toInt()
    val m = minutes.toInt()

    // Проверяем диапазоны
    return (h in 0..23) && (m in 0..59)
}

fun main() {
    print("Введите последовательность чисел: ")
    val input = readLine() ?: ""

    if (isValidTime(input)) {
        println("YES")
    } else {
        println("NO")
    }
}

File: TimeValidator/TimeValidator.cs
using System;

class Program
{
    static bool IsValidTime(string time)
    {
        // Должно быть ровно 4 цифры
        if (time.Length != 4)
        {
            return false;
        }

        // Первые 2 символа - часы, последние 2 символа - минуты
        string hours = time.Substring(0, 2);
        string minutes = time.Substring(2, 2);

        // Преобразуем строки в числа
        if (int.TryParse(hours, out int h) && int.TryParse(minutes, out int m))
        {
            // Проверяем диапазоны
            return (h >= 0 && h <= 23) && (m >= 0 && m <= 59);
        }

        return false;
    }

    static void Main()
    {
        Console.Write("Введите последовательность чисел: ");
        string input = Console.ReadLine();

        if (IsValidTime(input))
        {
            Console.WriteLine("YES");
        }
        else
        {
            Console.WriteLine("NO");
        }
    }
}
